extensions/vscode/src/views/homeView.ts

```diff
@@ -1179,15 +1204,14 @@ export class HomeViewProvider implements WebviewViewProvider, Disposable {
   ) => {
     try {
       const apis: Promise<void>[] = [this.refreshCredentialData()];
-      const selectedCR = await this.state.getSelectedContentRecord();
-      const selectedConfig = await this.state.getSelectedConfiguration();
-      if (!selectedCR || !selectedConfig || forceAll) {
-        // we have no current selection or it is invalid, so we have no choice but to
-        // search recursively for deployments
+      if (forceAll) {
+        // we have been told to refresh everything
         apis.push(this.refreshContentRecordData());
         apis.push(this.refreshConfigurationData());
       }
-      await Promise.all(apis);
+      const allApis = Promise.all(apis);
+      showProgress("Refreshing Data", allApis, Views.HomeView);
+      await allApis;
```

Comment:
Progress indicator is new... somehow we missed showing one here previously. It sure does help when the subdirectory is huge.